### PR TITLE
EPMRPP-34020 Implements generalTab component with redux-form.

### PR DESCRIPTION
**Pull request by ratmirj**

### CODE REVIEW

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    });
+  };
+
+  render() {
+    return (
+      <div className={styles.generalTabContainer}>
+        <form>
+          <div className={styles.formField}>
+            <div className={cellClasses}>
+              <label htmlFor={'name'}>Name</label>
```
Comment (greyd) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: Texts should be internationalized

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+  constructor(props) {
+    super(props);
+    this.state = props;
+  }
+
+  onTimeoutChange = (e) => {
+    this.setState({
+      inactivityTimeout: e,
+    });
+  };
```
Comment (greyd) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: there is a shorter version
`onTimeoutChange = (inactivityTimeout) => this.setState({ inactivityTimeout});`

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+import { BigButton } from '../../../../components/buttons/bigButton';
+
+const cx = classNames.bind(styles);
+const cellClasses = cx('gridCell', 'centered');
+const cellClassesSubmit = cx('gridCell', 'submitButton');
+const cellClassesFormField = cx('gridCell', 'formField');
+const cellClassesColumn = cx('gridCell', 'column');
+const cellClassesEmpty = cx('gridCell', 'empty');
+
+const optionsLaunch = [
```
Comment (greyd) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: Looks like a good candidate for excluding to a configuration file

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx (hunk 1 above)
Comment (greyd) [reply app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: Here and every not translated text below

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+            <div>
+              <div className={styles.formField}>
+                <div className={cellClasses}>
+                  <div>Base for Auto Analysis</div>
+                </div>
+
+                <div className={cellClasses}>
+                  <label className={cx('radioCheckbox')}>
+                    <div className={styles.CircleOuter}>
+                      {this.state.autoAnalysisBase === 'allLaunches' && (
```
Comment (greyd) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: Comparing with a 'magic string' can be risky. Any typo or a similar character in another language can be a problem that cannot be identified easily. I suggest using constants in all such places.

File: app/src/components/fields/fieldErrorHint/fieldErrorHint.scss [hunk 5]
```diff
@@ -2,6 +2,7 @@
   position: relative;
   text-align: left;
   height: 100%;
+  //width: 100%;
```
Comment (ArtsiomYeliseyenka) [inline app/src/components/fields/fieldErrorHint/fieldErrorHint.scss]: delete if this unnecessary.

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+    inactivityTimeout: PropTypes.string.isRequired,
+    keepLogs: PropTypes.string.isRequired,
+    keepAttachments: PropTypes.string.isRequired,
+    autoAnalysis: PropTypes.bool.isRequired,
+    submitFormData: PropTypes.func.isRequired,
+    autoAnalysisEnabled: PropTypes.bool,
+  };
+  render() {
+    const { autoAnalysisEnabled } = this.props;
+    const [BASE_ALL_LAUNCHES, BASE_SAME_NAME, BASE_CURRENT] = [
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: Move static data outside of render.

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+  form: 'testRunner',
+})
+@connect((state) => {
+  const autoAnalysisEnabled = selector(state, 'autoAnalysis');
+  return {
+    autoAnalysisEnabled,
+  };
+})
+export class GeneralTab extends Component {
+  static defaultProps = {
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: In accordance to project code style, we describe propTypes firstly, and then defaultProps.

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+import { timeStorageOptions, timeTimeoutOptions } from './timeOptions';
+import { FieldProvider } from '../../../../components/fields/fieldProvider';
+import { InputRadio } from '../../../../components/inputs/inputRadio';
+
+const cx = classNames.bind(styles);
+const cellClasses = cx('gridCell', 'centered');
+const cellClassesSubmit = cx('gridCell', 'submitButton');
+const cellClassesFormField = cx('gridCell', 'formField');
+const cellClassesColumn = cx('gridCell', 'column');
+const cellClassesEmpty = cx('gridCell', 'empty');
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: Describe classes directly in markup

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+                />
+              </label>
+            </div>
+            <div className={cellClassesFormField}>
+              <FieldProvider name="timeout">
+                <InputDropdown options={timeTimeoutOptions} />
+              </FieldProvider>
+            </div>
+            <div className={cellClasses}>
+              <span>
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: There is no need to add additional <span> element

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx (hunk 1 above)
Comment (ArtsiomYeliseyenka) [reply app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: +1

File: app/localization/translated/be.json [hunk 10]
```diff
@@ -132,6 +132,33 @@
   "ForgotPasswordForm.cancel": "Адмяніць",
   "ForgotPasswordForm.emailPlaceholder": "Увядзіце электронны адрас",
   "ForgotPasswordForm.sendEmail": "Адаслаць ліст",
+  "GeneralTab.autoAnalysis": "Auto analysis",
```
Comment (ArtsiomYeliseyenka) [inline app/localization/translated/be.json]: I think "ProjectSettingsGeneral. ...." is more suitable key name

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx [hunk 11]
```diff
@@ -0,0 +1,275 @@
+import React, { Component } from 'react';
+import classNames from 'classnames/bind';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { reduxForm, formValueSelector } from 'redux-form';
+import { connect } from 'react-redux';
+import { Input } from '../../../../components/inputs/input';
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: import { Input } from 'components/inputs/input'

Fix for all following cases. We use folder aliases on our project.

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx [hunk 12]
```diff
@@ -0,0 +1,275 @@
...
+
+@reduxForm({
+  form: 'testRunner',
+})
+@connect((state) => {
+  const autoAnalysisEnabled = selector(state, 'autoAnalysis');
+  return {
+    autoAnalysisEnabled,
+  };
+})
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: (state) => ({
    autoAnalysisEnabled: formValueSelector(FORM_ID)(state, 'autoAnalysis'),
  }),

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx [hunk 13]
```diff
@@ -0,0 +1,275 @@
...
+const cellClasses = cx('gridCell', 'centered');
+const cellClassesSubmit = cx('gridCell', 'submitButton');
+const cellClassesFormField = cx('gridCell', 'formField');
+const cellClassesColumn = cx('gridCell', 'column');
+const cellClassesEmpty = cx('gridCell', 'empty');
+
+const selector = formValueSelector('testRunner');
+
+@reduxForm({
+  form: 'testRunner',
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: Why "testRunner"?

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx [hunk 14]
```diff
@@ -0,0 +1,275 @@
...
+    return (
+      <div className={styles.generalTabContainer}>
+        <form>
+          <div className={styles.formField}>
+            <div className={cellClasses}>
+              <label htmlFor={'name'}>Name</label>
+            </div>
+            <div className={cellClasses}>
+              <FieldProvider name="name">
+                <Input id="name" disabled />
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: What the reason for provide id prop for Input component?

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx [hunk 15]
```diff
@@ -0,0 +1,275 @@
...
+                />
+              </label>
+            </div>
+            <div className={cellClassesFormField}>
+              <FieldProvider name="keepAttachments">
+                <InputDropdown options={timeStorageOptions} />
+              </FieldProvider>
+            </div>
+            <div className={cellClasses}>
+              <span>
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: span no needed here. Check for all following cases

File: app/src/pages/inside/settingsPage/generalTab/generalTab.jsx [hunk 16]
```diff
@@ -0,0 +1,275 @@
...
+                      <FormattedMessage
+                        id={'GeneralTab.autoAnalysisBase'}
+                        defaultMessage={'Base for Auto Analysis'}
+                      />
+                    </span>
+                  </div>
+                </div>
+
+                <div className={cellClasses}>
+                  <label className={cx('radioCheckbox')}>
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.jsx]: Please write all classnames in dash-notation. Check all following cases

File: app/src/pages/inside/settingsPage/generalTab/generalTab.scss [hunk 17]
```diff
@@ -0,0 +1,111 @@
+$border-color: #cccccc;
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.scss]: Use color-variables from colors.scss. There is no need to define additional.

File: app/src/pages/inside/settingsPage/generalTab/generalTab.scss [hunk 18]
```diff
@@ -0,0 +1,111 @@
...
+    &.empty {
+      display: none;
+    }
+    &.submitButton {
+      justify-content: flex-start;
+    }
+  }
+}
+
+.buttonWrapper {
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/generalTab/generalTab.scss]: All classes should be in dash-notation

**Pull request closed without merging**